packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -1724,9 +1724,9 @@ export class Bucket extends BucketBase {
     if (bucketName.length < 3 || bucketName.length > 63) {
       errors.push('Bucket name must be at least 3 and no more than 63 characters');
     }
-    const charsetMatch = bucketName.match(/[^a-z0-9.-]/);
+    const charsetMatch = bucketName.match(/[^a-z0-9._-]/);
```

Comment:
This method is used to validate the names of both imported buckets and buckets created in the app. But the underscore character should only be allowed when importing a bucket.

Reply:
Makes sense. I see that the pre-2018 bucket naming rules are as below:

```
Before March 1, 2018, buckets created in the US East (N. Virginia) Region could have names that were up to 255 characters long and included uppercase letters and underscores. Beginning March 1, 2018, new buckets in US East (N. Virginia) must conform to the same rules applied in all other Regions.
```

Seeing that this is a unique case only in us-east-1 and also has a further validation conditions like a longer length and capital letters being allowed, I do not think it makes sense to add a very specific validation check for this scenario. Closing the PR.